### ISSUE TITLE
Surrounding font size not determined correctly

### ISSUE THREAD

**Issue by mhchem** [mathjax/MathJax#1535]
Sometimes, the font-size is determined wrongly. Here, two similar fonts, served via Google Fonts, result in very different sizing of MathJax. I don't know how you determine the text height. This might be a badly implemented font.

```
<!doctype html>
<html>
<head>
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>
<link href="https://fonts.googleapis.com/css?family=Alegreya|Crimson+Text" rel="stylesheet">
</head>
<body style="font-size:20px">
<div>This is an \(x\).</div>
<div style="font-family:Alegreya">This is an \(x\).</div>
<div style="font-family:'Crimson Text'">This is an \(x\).</div>
</body>
</html>
```

**Comment by dpvc** [comment]
You are right that the second font is at fault.  MathJax uses the fonts ex-height parameter to determine the size of its own fonts.  This value is supposed to give the height of the lower-case letters in the font (the height of an "x").  But the Crimson Text ex-height is incorrectly set, as can be seen using the following file

```
<!doctype html>
<html>
<head>
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>
<link href="https://fonts.googleapis.com/css?family=Alegreya|Crimson+Text" rel="stylesheet">
<style>
.exbox {
  display: inline-block;
  width: 1ex;
  height:1ex;
  background-color:red
}
.embox {
  display: inline-block;
  width: 1ex;
  height:1em;
  background-color:green
}
</style>
</head>
<body style="font-size:20px">
<div>This is an \(x\)<span class="exbox"></span><span class="embox"></span>.</div>
<div style="font-family:Alegreya">This is an \(x\)<span class="exbox"></span><span class="embox"></span>.</div>
<div style="font-family:'Crimson Text'">This is an \(x\)<span class="exbox"></span><span class="embox"></span>.</div>
</body>
</html>
```

which produces 

<img width="158" alt="ex-heights" src="https://cloud.githubusercontent.com/assets/432782/16283485/f58dc178-389a-11e6-996d-d09134be9443.png">

Here the red squares are 1ex on a side, and the green boxes are 1em high.  As you can see, the last font's ex-height is considerably too small (and its em-size is rather larger than necessary). My experience with Google fonts is that many of them have bad metric information, and that the ex-height frequently is incorrect, in particular.

There is also another issue that can occur with the web fonts; since MathJax must measure the ex- and em-heights empirically, this only works properly when the font is fully loaded from the server.  Since web fonts come asynchronously, MathJax may have measured the temporary font used by the browser before the font arrives.  (That's not the case, here, but it certainly can be a problem.)  So if you are using web fonts, you either need to delay MathJax's initial typesetting until all the fonts are in place, or use the MatchWebFonts extension (a hack that re-renders expressions if the surrounding font changes due to a wen font arriving).
